src/main/java/Deadline.java

```diff
@@ -0,0 +1,84 @@
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+    private String deadLine;
+    private String interpretedDate;
+    private String interpretedTime;
+
+    /**
+     * @param taskName task of the Deadline object
+     * @param deadLine deadline of the Deadline object in the following format: 2/12/2019 1800
+     *
+     */
+
+
+    public Deadline(String taskName, String deadLine) {
+        super(taskName);
+        this.deadLine = deadLine;
+
+        interpretDeadLine();
+
+    }
+
+    /**
+     * returns the message comprising [D], name of the task of the deadline object and its deadline,
+     * together with the a symbol representing if a deadline object is completed or not.
+     * [D] represents that our object is a deadline object.
+     *
+     */
+    public String toString() {
+        String box;
+        String msg = "[D]";
+        if (!this.isDone()) {
+            box = "[✗]";
+        } else {
+            box = "[✓]";
+        }
+        return msg + box + " " + this.getTask() + " (by: " + this.interpretedDate + " at " + this.interpretedTime + ")";
+    }
+
+    /**
+     * interprets deadline of Deadline object.
+     *eg 2/12/2019 becomes 2 December 2019
+     *and 1830 becomes 0630 pm
+     */
+
+    public void interpretDeadLine() {
```

Comment:
Would it be more appropriate for Parser class to handle processing of input dates? Besides the idea of Parser being the class that manages input information, Event also interprets dates and letting Parser handle for both Deadline and Event reduces duplicate code.